### PR TITLE
ADP - add missing ProvisioningState

### PR DESCRIPTION
**Pull request by aelij**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by aelij** [comment]
Since this is a new unpublished RP, the breaking change doesn't matter. The [original PR](https://github.com/Azure/azure-rest-api-specs/pull/10925) was merged earlier today.

**Comment by lirenhe** [comment]
@aelij, it seems your change contains breaking change, could you follow the process from http://aka.ms/bcforapi to review this breaking change?

File: specification/adp/resource-manager/Microsoft.AutonomousDevelopmentPlatform/preview/2020-07-01-preview/adp.json [hunk 1]
```diff
@@ -674,6 +674,7 @@
             "Succeeded",
             "Failed",
             "Canceled",
+            "Accepted",
```
Comment (johanste) [inline specification/adp/resource-manager/Microsoft.AutonomousDevelopmentPlatform/preview/2020-07-01-preview/adp.json:677]: Why is this enum marked as `"modelAsString": false`?

Comment (aelij) [reply specification/adp/resource-manager/Microsoft.AutonomousDevelopmentPlatform/preview/2020-07-01-preview/adp.json:677]: I followed the same pattern for `ProvisioningState` from other RPs. Why do you think it should it be `true`? I would like an enum to be generated in the SDK.

Comment (johanste) [reply specification/adp/resource-manager/Microsoft.AutonomousDevelopmentPlatform/preview/2020-07-01-preview/adp.json:677]: Closed enums (modelAsString: false) have proven to be version intolerant (over time). If you can commit to never, ever add a new provisioning state value, then this is ok.

**Review by lirenhe — APPROVED** [review]

**Pull request merged**